huaweicloud/services/servicestage/data_source_huaweicloud_servicestage_component_runtimes.go

```diff
@@ -0,0 +1,100 @@
+package servicestage
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk/openstack/servicestage/v2/metadata"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/helper/hashcode"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+)
+
+func DataSourceComponentRuntimes() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceComponentRuntimesRead,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"default_port": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"runtimes": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"default_port": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceComponentRuntimesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	region := config.GetRegion(d)
+	client, err := config.ServiceStageV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating ServiceStage v2 client: %s", err)
+	}
+
+	resp, err := metadata.ListRuntimes(client)
+	if err != nil {
+		return diag.Errorf("error retrieving the runtimes list: %s", err)
+	}
+
+	filter := map[string]interface{}{
+		"Type":                 d.Get("name"),
+		"ContainerDefaultPort": d.Get("default_port"),
+	}
+	filtResult, err := utils.FilterSliceWithField(resp, filter)
+	if err != nil {
+		return fmtp.DiagErrorf("filter CSS flavors failed: %s", err)
```

Comment:
1 . s/fmtp.diagErrorf/diag.Errorf
2. this is not CSS flavors

Reply:
done.